Source/santasyncservice/SNTPushClientNATS.mm

```diff
@@ -0,0 +1,765 @@
+@property(nonatomic)
+    NSMutableArray<NSValue *> *tagSubscriptions;        // Array of natsSubscription pointers
+@property(nonatomic) dispatch_queue_t connectionQueue;  // Single queue for connection management
+@property(nonatomic) dispatch_queue_t messageQueue;     // Queue for processing messages
+@property(nonatomic, readwrite) BOOL isConnected;
+@property(nonatomic, readwrite) NSUInteger fullSyncInterval;
+@property(atomic) BOOL isShuttingDown;
+// Push notification configuration from preflight
+@property(nonatomic, copy) NSString *pushServer;
+@property(nonatomic, copy) NSString *pushToken;  // nkey
+@property(nonatomic, copy) NSString *jwt;
+@property(nonatomic, copy) NSString *pushDeviceID;
+@property(nonatomic, copy) NSArray<NSString *> *tags;
+@property(nonatomic) BOOL hasSyncedWithServer;
+// Connection retry state
+@property(nonatomic) dispatch_source_t connectionRetryTimer;
+@property(nonatomic) NSTimeInterval currentRetryDelay;
+@property(nonatomic) NSInteger retryAttempt;
+@property(nonatomic) BOOL isRetrying;
+@end
+
+@implementation SNTPushClientNATS
+
+- (instancetype)initWithSyncDelegate:(id<SNTPushNotificationsSyncDelegate>)syncDelegate {
+  self = [super init];
+  if (self) {
+    _syncDelegate = syncDelegate;
+    _fullSyncInterval = kDefaultPushNotificationsFullSyncInterval;
+    _connectionQueue =
+        dispatch_queue_create("com.northpolesec.santa.nats.connection", DISPATCH_QUEUE_SERIAL);
+    _messageQueue =
+        dispatch_queue_create("com.northpolesec.santa.nats.message", DISPATCH_QUEUE_SERIAL);
+    _isShuttingDown = NO;
+    _hasSyncedWithServer = NO;
+    _tagSubscriptions = [NSMutableArray array];
+    _currentRetryDelay = 1.0;  // Start with 1 second
+    _retryAttempt = 0;
+    _isRetrying = NO;
+
+    // Don't connect immediately - wait for preflight to provide configuration
+    LOGI(@"NATS push client: Initialized, waiting for preflight configuration");
+  }
+  return self;
+}
+
+- (void)dealloc {
+  // Don't call disconnect here to avoid race conditions
+  // Cleanup should be done explicitly before dealloc
+  if (self.conn || self.deviceSub || self.globalSub) {
+    LOGW(@"NATS: Client deallocated without proper disconnect");
+  }
+}
+
+- (void)configureWithPushServer:(NSString *)server
+                      pushToken:(NSString *)token
+                            jwt:(NSString *)jwt
+                   pushDeviceID:(NSString *)deviceID
+                           tags:(NSArray<NSString *> *)tags {
+  dispatch_async(self.connectionQueue, ^{
+    if (self.isShuttingDown) return;
+
+    NSString *fullServer;
+#ifdef SANTA_FORCE_SYNC_V2
+    // In debug builds, allow overriding the domain suffix
+    fullServer = server;
+    LOGW(@"NATS: Domain suffix disabled for debugging - using server as-is: %@", fullServer);
+#else
+    // Always append domain suffix in release builds 
+    fullServer = [NSString stringWithFormat:@"%@.push.northpole.security", server];
```

Comment:
The push_server field in preflight has the hostname, this should check that it contains push.northpole.security instead of always adding it.